src/VisualStudio/Core/Impl/ProjectSystem/CPS/CPSProject_IWorkspaceProjectContext.cs

```diff
@@ -59,22 +59,36 @@ bool IWorkspaceProjectContext.LastDesignTimeBuildSucceeded
                 _lastDesignTimeBuildSucceeded = value;
             }
         }
+
+        string IWorkspaceProjectContext.BinOutputPath
+        {
+            get
+            {
+                return this.TryGetBinOutputPath();
+            }
+            set
+            {
+                SetOutputPathAndRelatedData(objOutputPath: this.TryGetObjOutputPath(), binOutputPath: value);
```

Comment:
Why is this called TryGetBinOutputPath and TryGetObjOutputPath? They are both just returning fields - why not just use the fields? They also don't match the Try pattern.